File: 2-Patron/php/cliente.php
<?php
class Cliente
{
  private $nombre;
  private $apellido;
  private $edad;
  private $correo;
  private $contrasena;

  function __construct($nombre,$apellido,$edad,$correo,$contrasena)
  {
    $this->$nombre = $nombre;
    $this->$apellido = $apellido;
    $this->$edad = $edad;
    $this->$correo = $correo;
    $this->$contrasena = $contrasena;
  }

  function getNombre()
  {
    return $this->nombre;
  }

  function setNombre($nombre)
  {
    $this->nombre = $nombre;
  }

  function getApellido()
  {
    return $this->apellido;
  }

  function setApellido($apellido)
  {
    $this->apellido = $apellido;
  }

  function getEdad()
  {
    return $this->edad;
  }

  function setEdad($edad)
  {
    $this->edad = $edad;
  }

  function getCorreo()
  {
    return $this->correo;
  }

  function setNombre($correo)
  {
    $this->correo = $correo;
  }

  function getContra()
  {
    return $this->contraseña;
  }

  function setContra($contraseña)
  {
    $this->contraseña = $contraseña;
  }

  function autenticar()
  {
    //
  }
}
?>

File: 2-Patron/Java/Cliente.java
package abstractmasterwhole;

public class Cliente {
    private String nombre;
    private String apellido;
    private int edad;
    private String correo;
    private String contrasena;

    public Cliente(String nombre, String apellido, int edad, String correo, String contrasena) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.edad = edad;
        this.correo = correo;
        this.contrasena = contrasena;
    }

    public String getApellido() {
        return apellido;
    }

    public String getContrasena() {
        return contrasena;
    }

    public String getCorreo() {
        return correo;
    }

    public int getEdad() {
        return edad;
    }

    public String getNombre() {
        return nombre;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public void setContrasena(String contrasena) {
        this.contrasena = contrasena;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    public void autenticar(){
        
    }
}
